cmd/rpcdaemon/commands/debug_api.go

```diff
@@ -118,33 +117,45 @@ func (api *PrivateDebugAPIImpl) AccountRange(ctx context.Context, blockNrOrHash
 	return res, nil
 }
 
-func (api *PrivateDebugAPIImpl) GetModifiedAccountsByNumber(_ context.Context, startNum rpc.BlockNumber, endNum *rpc.BlockNumber) ([]common.Address, error) {
-	if endNum != nil && startNum.Int64() >= endNum.Int64() {
-		return nil, fmt.Errorf("start block height (%d) must be less than end block height (%d)", startNum.Int64(), endNum.Int64())
+// GetModifiedAccountsByNumber returns a list of accounts found in the change sets
+// startNum - first block from which to include results
+// endNum - if present, last block from which to include results (inclusive). If not present, startNum.
+func (api *PrivateDebugAPIImpl) GetModifiedAccountsByNumber(ctx context.Context, startNumber rpc.BlockNumber, endNumber *rpc.BlockNumber) ([]common.Address, error) {
+	tx, err := api.db.Begin(ctx, nil, false)
+	if err != nil {
+		return nil, err
 	}
+	defer tx.Rollback()
 
-	final, _, err := stages.GetStageProgress(api.dbReader, stages.Finish)
+	latestBlock, _, err := stages.GetStageProgress(tx, stages.Finish)
 	if err != nil {
 		return nil, err
 	}
 
-	lastBlockNumber := final
+	// forces negative numbers to fail (too large) but allows zero
+	startNum := uint64(startNumber.Int64())
+	if startNum > latestBlock {
+		return nil, fmt.Errorf("start block (%d) is later than the latest block (%d)", startNum, latestBlock)
+	}
 
-	if startNum.Int64() < 1 || uint64(startNum.Int64()) > lastBlockNumber {
-		return nil, fmt.Errorf("start block %x not found", uint64(startNum.Int64()))
+	endNum := startNum // allows for single param calls
+	if endNumber != nil {
+		// forces negative numbers to fail (too large) but allows zero
+		endNum = uint64(endNumber.Int64())
 	}
 
-	if endNum == nil {
-		*endNum = startNum
-	} else {
-		if endNum.Int64() < 1 || uint64(endNum.Int64()) > lastBlockNumber {
-			return nil, fmt.Errorf("end block %x not found", uint64(endNum.Int64()))
-		}
+	// is endNum too big?
+	if endNum > latestBlock {
+		return nil, fmt.Errorf("end block (%d) is later than the latest block (%d)", endNum, latestBlock)
 	}
 
-	return getModifiedAccounts(api.dbReader, uint64(startNum.Int64()), uint64(endNum.Int64()))
+	// will check that startNum is less or equal to endNum
+	return getModifiedAccounts(api.dbReader, startNum, endNum)
```

Comment:
Can you try to change this to pass `tx` too? Do you feel adventurous enough? I can help you. You need to modify `eth.GetModifiedAccounts` function (in `ethdb/walk.go`) from using `Walk` function to use a cursor and for loop instead

Reply:
I'm totally adventurous. I'll try it now.

Reply:
Should I try to fix it elsewhere as well or make a new PR?

For example, here: https://github.com/ledgerwatch/turbo-geth/blob/master/cmd/rpcdaemon/commands/eth_api.go#L146

Reply:
Maybe make a new PR for other things